great_expectations/core/validation_config.py

```diff
@@ -19,30 +67,139 @@ class ValidationConfig(BaseModel):
         name: The name of the validation.
         data: A batch config to validate.
         suite: A grouping of expectations to validate against the data.
+        id: A unique identifier for the validation; added when persisted with a store.
 
     """
 
     class Config:
-        arbitrary_types_allowed = True
+        arbitrary_types_allowed = (
+            True  # Necessary for compatibility with suite's Marshmallow dep
+        )
+        """
+        When serialized, the suite and data fields should be encoded as a set of identifiers.
+        These will be used as foreign keys to retrieve the actual objects from the appropriate stores.
+
+        Example:
+        {
+            "name": "my_validation",
+            "data": {
+                "datasource": {
+                    "name": "my_datasource",
+                    "id": "a758816-64c8-46cb-8f7e-03c12cea1d67"
+                },
+                "asset": {
+                    "name": "my_asset",
+                    "id": "b5s8816-64c8-46cb-8f7e-03c12cea1d67"
+                },
+                "batch_config": {
```

Comment:
Yes because the datasource is the top-level domain object that has an applicable store, we need to use it as our entry point in order to know which asset and batch config to grab